web/pdf_page_view.js

```diff
@@ -538,22 +538,24 @@ var PDFPageView = (function PDFPageViewClosure() {
       var viewport = pdfPage.getViewport(1);
       // Use the same hack we use for high dpi displays for printing to get
       // better output until bug 811002 is fixed in FF.
-      var PRINT_OUTPUT_SCALE = 2;
+      var DPI = 200; // increase to improve quality
+      var PRINT_OUTPUT_SCALE = DPI/72; 
```

Comment:
The memory footprint is definitely a concern.  I'm not sure yet if that highlights a problem with the way this patch was implemented, or a symptom of a larger problem with the way printing is handled in general.  Printing text as an image is different than printing text with font information.

I'll try to see if keeping the scale factor of 2 as it was before (DPI = 144) will keep the memory footprint the same as before (I suspect it will).

With a DPI setting of 500 and printing an image, the quality of print is virtually indistinguisable from printing a PDF with adobe reader which uses Fonts.